src/components/ToDoList/ToDoList.js

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import ToDoListItem from "../ToDoListItem";
+import "./ToDoList.scss";
+import { useSelector, useDispatch } from "react-redux";
+import {
+  deleteElement,
+  checkCompleted,
+  reorderElements,
+} from "../../redux/toDoListSlice";
+import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd";
+
+const ToDoList = () => {
+  const elements = useSelector((state) => state.toDoList.elements);
+  const filter = useSelector((state) => state.toDoList.filter);
+  const dispatch = useDispatch();
+
+  const filterElements = (elements, filter) => {
+    switch (filter) {
+      case "all": {
+        return elements;
+      }
+      case "active": {
+        return elements.filter((el) => !el.completed);
+      }
+      case "completed": {
+        return elements.filter((el) => el.completed);
+      }
+      default: {
+        return elements;
+      }
+    }
+  };
+
+  const handleOnDragEnd = (result) => {
+    dispatch(reorderElements(result));
+  };
+
+  const filteredElements = filterElements(elements, filter);
+
+  const list = filteredElements.map((el) => {
+    const { id } = el;
+
+    const index = elements.findIndex((el) => el.id === id);
+
+    const onDeleted = (id) => {
```

Comment:
There is no need to create function on each iteration. Just move it outside `map`

Reply:
resolved